pkg/install/1-installresources_test.go

```diff
@@ -0,0 +1,70 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"testing"
+
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+	aztypes "github.com/openshift/installer/pkg/types/azure"
+)
+
+func TestZones(t *testing.T) {
+	tests := []struct {
+		zones      []string
+		replicas   *int64
+		shouldPass bool
```

Comment:
reverse the sense of this.  Write `wantErr string` and check the error message matches the expectation - see other cases where the same is done.